lib/server/services/qualityControlFlag/QcFlagService.js

```diff
@@ -439,17 +439,22 @@ class QcFlagService {
             await QcFlagRepository.removeOne({ where: { id } });
 
             /*
-             * Restore default effective periods of older flags in order to make update procedure working.
+             * Remove all effective periods before restoring default once of older flags in order to make update procedure working.
              * It works only in one direction (reducing effective periods) by removing, shorthening, splitting effective periods.
              */
             await QcFlagEffectivePeriodRepository.removeAll({
                 where: { flagId: { [Op.in]: flagsCreatedBeforeRemovedFlag.map(({ id }) => id) } },
             });
-            await QcFlagEffectivePeriodRepository.insertAll(flagsCreatedBeforeRemovedFlag
-                .map(({ id, from, to }) => ({ flagId: id, from, to })));
 
             // Update effective periods of flags created before removed flag
             for (const newestFlagToBeUpdatedIndex in flagsCreatedBeforeRemovedFlag) {
+                const newestFlagToBeUpdated = flagsCreatedBeforeRemovedFlag[newestFlagToBeUpdatedIndex];
```

Comment:
Why is it newest if it loops over `flags created before` ?
The comment means update but what is done is insertion?

Reply:
1. because it's the newest flag that is updated during given loop
2. insertion + update - changed to `restore effective periods ...`